scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextVariableExtractor.java

```diff
@@ -122,11 +122,17 @@
 
     public Map<String, Serializable> extractScopeVariables(TaskContext taskContext) {
         Map<String, Serializable> variables = new HashMap<>();
-
+        Map<String, Serializable> previousVariables = null;
```

Comment:
This will throw a null pointer exception on line 135. 
Better to always initialise the maps:

Map<String, Serializable> previousVariables = new HashMap<>();

Reply:
Agree